packages/core/lib/segments/segment.d.ts

```diff
@@ -0,0 +1,59 @@
+import Subsegment = require('./attributes/subsegment');
+import IncomingRequestData = require('../middleware/incoming_request_data');
+
+declare class Segment {
+  id: string;
+  name: string;
+  start_time: number;
+  in_progress?: boolean;
+  counter: number;
```

Comment:
this is an example of where I wonder, should `counter` be public?

Reply:
You are correct, it could be a private attribute.